app/src/layouts/timeline/index.ts

```diff
@@ -0,0 +1,312 @@
+					(relation) =>
+						relation.collection === props.collection &&
+						relation.field === field.field &&
+						relation.related_collection === 'directus_users'
+				);
+
+				return related !== undefined;
+			},
+		});
+
+		const { title, dateField, timeField, userField } = useLayoutOptions();
+		const { sort, page, fields } = useLayoutQuery();
+
+		const { days, loading } = useDays();
+
+		const timeFieldRequired = computed(() => {
+			const field = fieldsInCollection.value.find((field) => field.field === dateField.value);
+			return field?.type === 'date';
+		});
+
+		return {
+			page,
+			fieldsInCollection,
+			primaryKeyField,
+			title,
+			sort,
+			info,
+			dateField,
+			timeField,
+			userField,
+			fieldGroups,
+			timeFieldRequired,
+			days,
+			useEvents,
+			loading,
+		};
+
+		function parseUrl(file: Record<string, any>) {
+			if (!file || !file.type) return null;
+			if (file.type.startsWith('image') === false) return null;
+			if (file.type.includes('svg')) return null;
+
+			const key = 'system-medium-cover';
+
+			const url = getRootPath() + `assets/${file.id}?key=${key}&modified=${file.modified_on}`;
+			return addTokenToURL(url);
+		}
+
+		function useLayoutOptions() {
+			const dateField = createViewOption<string | null>('dateField', fieldGroups.value.date?.[0]?.field ?? null);
+			const timeField = createViewOption<string | null>('timeField', null);
+			const userField = createViewOption<string | null>('userField', fieldGroups.value.user?.[0]?.field ?? null);
+			const title = createViewOption<string | null>('title', null);
+
+			return { dateField, timeField, userField, title };
+
+			function createViewOption<T>(key: keyof LayoutOptions, defaultValue: any) {
+				return computed<T>({
+					get() {
+						return layoutOptions.value?.[key] !== undefined ? layoutOptions.value?.[key] : defaultValue;
+					},
+					set(newValue: T) {
+						layoutOptions.value = {
+							...layoutOptions.value,
+							[key]: newValue,
+						};
+					},
+				});
+			}
+		}
+
+		function useDays() {
+			const days = ref<Day[]>([]);
+			const loading = ref(false);
+
+			const daysFilter = computed(() => {
+				if (!dateField.value) {
+					return filter.value;
+				}
+
+				const daysRange = {
+					[dateField.value]: {
+						_between: [`$NOW(${(page.value - 2) * 30 + 1} days)`, `$NOW(${(page.value - 1) * 30} days)`],
+					},
+				};
+
+				return {
+					_and: [filter.value, daysRange],
+				};
+			});
+
+			watch(
+				[page, collection, daysFilter],
+				() => {
+					fetchDays();
+				},
+				{ immediate: true }
+			);
+
+			return { days, fetchDays, loading };
+
+			async function fetchDays() {
+				loading.value = true;
+
+				const result = await api.get(getEndpoint(collection.value!), {
+					params: {
+						groupBy: [`year(${dateField.value})`, `month(${dateField.value})`, `day(${dateField.value})`],
+						aggregate: {
+							count: '*',
+						},
+						limit: -1,
+						// sort: sort.value,
```

Comment:
![WindowsTerminal_l9reRopU5W](https://user-images.githubusercontent.com/42867097/143547033-fdba2a6f-0925-4d92-8398-04a3479e6bbc.png)

enabling sort here seems to trigger the similar error in #10050, where `order by "tasks"."end_date" desc` in turn necessitates `end_date` to be added to groupBy. Adding it does resolve it, but the grouping by end_date ended up messing up the group instead. Not sure what's happening here.